controllers/commerce.js

```diff
@@ -0,0 +1,343 @@
+    const operatingUnitNumber = "277";
+
+    const myContext = {
+            requestContext: 
+            {
+                apiSettings: {
+                    baseUrl: "",
+                    channelId: channelId,
+                    oun: operatingUnitNumber,
+                    baseImageUrl: "",
+                },
+                user: {
+                    token: "string",
+                    isAuthenticated: true,
+                },
+                locale: "en-gb",
+                operationId: "", 
+                query: undefined, //IDictionary<string>
+                params: undefined, //IRequestContextParams;
+                accessToken: accessToken,
+            },
+        };
+    
+        try {
+            // Investigation of methods to fetch product information from Commerce server
+
+            // Method 1: Fetch details about a single product by ID
+            console.log('Calling getByIdAsync for recordId:', recordId);
+            
+            const product = await getByIdAsync(myContext, recordId, channelId);
+           
+            console.log("Name: ", product.Name, " ItemId: ", product.ItemId , " Price: " , product.Price);
+
+            // Method 2: Fetch details about a group of products using IDs
+            const products = await getByIdsAsync(myContext, channelId, [68719552888,68719552889,68719552890], 
+                                                 null,  //inventLocationId?: string |
+                                                 null); //catalogId?: number | null)
+
+            console.log('Product getByIdsAsync:', JSON.stringify(products));
+
+            products.forEach(function (product){
+                console.log('RecordId:', product.RecordId);
+                console.log('ProductNumber:', product.ProductNumber);
+                console.log('ItemId:', product.ItemId);
+                console.log('Name:', product.Name);
+                console.log('Price:', product.Price);
+                console.log('MasterProductId:', product.MasterProductId);
+            });
+
+            // Method 3: Using product search API
+
+            /* Rules deduced from error messages:
+            1.  This criteria combination is not suppoted: 
+                useProductIds=True; useCategoryIds=False; useItemIds=True; useKeywords=False. 
+                Only one search criteria type can be specified.
+        
+            2.  When searching virtual catalogue products, you can search only by id, category or keyword.
+            */
+
+            var searchResult;
+            var myProductSearchCriteria;
+
+            /* Search by Category ID */
+            myProductSearchCriteria = { 
+                //Ids: [],
+                //Ids?: number[];
+                CategoryIds: [68719491845],  // from category hierarchy - record ID
+                //IncludeProductsFromDescendantCategories: true, //IncludeProductsFromDescendantCategories?: boolean;
```

Comment:
Potentially keeping this code commented for future reference, in which case it's fine. Otherwise might want to tidy this up